.github/scripts/check-via-duplication.py

```
@@ -0,0 +1,107 @@
+    for scope in ROOTS:
+        scoped = [row for row in records if row[0][0] == scope]; print(f"| {scope} | {len(scoped)} | {sum(key[0] == scope for key in debt)} | {sum(row[5] for row in scoped)} |")
+    print("\n## Pair debt\n\n| Scope | Core path | Verifier path | Tokens | Disposition |\n|---|---|---|---:|---|")
+    for key, value in sorted(debt.items(), key=lambda item: (-item[1], item[0])): print(f"| {key[0]} | `{key[1]}` | `{key[2]}` | {value} | {'bounded-allowlist' if key in allowed else 'actionable'} |")
+    for title, selected in (("Actionable", False), ("Bounded-allowlist", True)):
+        print(f"\n## {title} clone records\n"); matches = [row for row in records if (row[0] in allowed) is selected]
+        if not matches: print("_None._")
+        for key, cs, ce, vs, ve, tokens, lines in matches: print(f"- `{key[0]}` `{key[1]}:{cs}-{ce}` ↔ `{key[2]}:{vs}-{ve}` - {tokens} tokens, {lines} lines; `{'bounded-allowlist' if selected else 'actionable'}`")
+    actionable = [(key, value) for key, value in sorted(debt.items(), key=lambda item: (-item[1], item[0])) if key not in allowed]
+    print("\n## Issue #382 actionable pair work-list\n\n" + ("\n".join(f"- [ ] `{key[1]}` ↔ `{key[2]}` - {value} duplicate tokens" for key, value in actionable) or "_None._"))
+    metadata = (ROOT / ".github/sibling-paths.yml").read_text(); pair_lines = [line for line in metadata.splitlines() if line.lstrip().startswith("- ")]
+    registered = {tuple(value.strip() for value in pair) for pair in re.findall(r"- \[([^,]+), ([^\]]+)\]", metadata)}; require(registered and len(registered) == len(pair_lines), "could not read every registered sibling family")
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)